src/libecalc/domain/process/core/stream/stream.py

```diff
@@ -0,0 +1,208 @@
+from __future__ import annotations
+
+from dataclasses import dataclass
+from functools import cached_property
+from typing import ClassVar, Protocol
+
+from libecalc.common.units import UnitConstants
+from libecalc.domain.process.core.stream.conditions import ProcessConditions
+from libecalc.domain.process.core.stream.fluid import Fluid
+from libecalc.domain.process.core.stream.mixing import SimplifiedStreamMixing
+
+
+class StreamMixingStrategy(Protocol):
+    """Protocol for stream mixing strategies"""
+
+    def mix_streams(self, streams: list[Stream]) -> Stream:
+        """Mix multiple streams into a single resultant stream"""
+        ...
+
+
+@dataclass
+class Stream:
+    """
+    Represents a fluid stream with its properties and conditions.
+
+    This class consolidates the functionality from various existing stream
+    implementations in the system.
+
+    Attributes:
+        fluid: Fluid object containing composition and EoS model
+        conditions: Process conditions (temperature and pressure)
+        mass_rate: Mass flow rate [kg/h]
+        name: Optional identifier for the stream
+    """
+
+    fluid: Fluid
+    conditions: ProcessConditions
+    mass_rate: float
+
+    # Default mixing strategy (class variable)
+    _stream_mixing_strategy: ClassVar[StreamMixingStrategy] = SimplifiedStreamMixing()
+
+    def __post_init__(self):
+        """Validate stream properties"""
+        if self.mass_rate < 0:
+            raise ValueError(f"Mass rate must be non-negative, got {self.mass_rate}")
```

Comment:
Should probably be a Stream specific exception, easier to capture it. InvalidstreamException as parent, `NegativeMassRateException` in this case? Or something like that